### PR TITLE
chore(pkg/controllers): remove unused secret and no-op comparisons

### PR DESCRIPTION
**Pull request by uhthomas**
Fixes: #167

### CODE REVIEW

File: pkg/controllers/wireguard_controller.go [hunk 1]
```diff
@@ -460,8 +460,6 @@ func (r *WireguardReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 			return ctrl.Result{}, err
 		}
 
-		bytes.Equal(b, secret.Data["state.json"])
```
Comment (uhthomas) [inline pkg/controllers/wireguard_controller.go:463]: Any idea what the intention of this was?

File: pkg/controllers/wireguard_controller.go [hunk 2]
```diff
@@ -519,33 +517,14 @@ func (r *WireguardReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 			return ctrl.Result{}, err
 		}
 
-		bytes.Equal(b, secret.Data["state"])
```
Comment (uhthomas) [inline pkg/controllers/wireguard_controller.go:522]: Same here @jodevsa

File: pkg/controllers/wireguard_controller.go (hunk 1 above)
Comment (jodevsa) [reply pkg/controllers/wireguard_controller.go:463]: haha probably a mistake :D

File: pkg/controllers/wireguard_controller.go (hunk 2 above)
Comment (jodevsa) [reply pkg/controllers/wireguard_controller.go:522]: 😓 not sure what I was doing

**Review by jodevsa — APPROVED** [review]
LGTM, thanks a lot!

**Pull request merged**